core/textile/client.go

```diff
@@ -232,16 +304,17 @@ func (tc *textileClient) IsRunning() bool {
 	return tc.isRunning
 }
 
-func (tc *textileClient) getThreadContext(parentCtx context.Context, threadName string, dbID thread.ID) (context.Context, error) {
+func (tc *textileClient) getThreadContext(parentCtx context.Context, threadName string, dbID thread.ID, hub bool) (context.Context, error) {
 	var err error
 	ctx := parentCtx
 
 	if err = tc.requiresRunning(); err != nil {
 		return nil, err
 	}
 
-	// If we are connected to the Hub, add the keys to the context so we can replicate
-	if tc.isConnectedToHub == true {
+	// Some threads will be on the hub and some will be local, this flag lets you specify
+	// where it is, perhaps an improvement could be to save this flag in the store and load it from there
+	if hub {
```

Comment:
For 1 specifically I thought the metathread would be local but replicated without exposing the data to the hub (so it could be restored later), this way it will still work fully offline. If a hub connected is present when creating the metathread, but then they later go offline it wouldn't work because it wouldn't be able to find even local buckets.  Everything else makes sense to me so will leave it as a flag for the function and remove the comment.

Reply:
Gotcha, makes sense 👍

Reply:
Just cleaned up the comment for now. Didn't add that field to the bucket schema in the metathread yet because all created buckets are local right now (shared buckets aren't added to metathread since it will come from a new shared with me thread).